.github/workflows/check-links.yml

```diff
@@ -0,0 +1,44 @@
+# This action requires that any PR targeting the main branch should touch at
+# least one CHANGELOG file. If a CHANGELOG entry is not required, add the "Skip
+# Changelog" label to disable this action.
+
+name: 'Check Links'
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+  schedule:
+    - cron: "0 4 * * 1" # Run at 04:00 UTC on Mondays.
+
+env:
+  # renovate: datasource=golang-version depName=go
+  GO_VERSION: "1.23.6"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
+
+jobs:
+  check-links:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Link Checker
+        id: lychee
+        uses: lycheeverse/lychee-action@f613c4a64e50d792e0b31ec34bbcbba12263c6a6 # v2.3.0
+        with:
+          fail: ${{ github.event_name != 'schedule' }}
+
+      - name: Create Issue From File
+        if: steps.lychee.outputs.exit_code != 0 && github.event_name == 'schedule'
+        uses: peter-evans/create-issue-from-file@e8ef132d6df98ed982188e460ebb3b5d4ef3a9cd # v5.0.1
+        with:
+          title: Lychee found some broken links
+          content-filepath: ./lychee/out.md
+          labels: bug
```

Comment:
I'd also be fine with the build failing on the next PR (or main if we keep the schedule) in case you'd like to avoid pulling in another GH workflow action.

Reply:
no i think that's fine, better to get notified when something is wrong